packages/EasyCore/src/Doctrine/Dispatchers/DeferredEntityEventDispatcher.php

```diff
@@ -39,90 +44,123 @@ public function __construct(EventDispatcherInterface $eventDispatcher)
     public function clear(?int $transactionNestingLevel = null): void
     {
         if ($transactionNestingLevel !== null) {
-            foreach (\array_keys($this->entityInsertions) as $level) {
+            foreach (\array_keys($this->entityChangeSets) as $level) {
                 if ($level >= $transactionNestingLevel) {
-                    $this->entityInsertions[$level] = [];
-                }
-            }
-
-            foreach (\array_keys($this->entityUpdates) as $level) {
-                if ($level >= $transactionNestingLevel) {
-                    $this->entityUpdates[$level] = [];
+                    $this->entityChangeSets[$level] = [];
                 }
             }
 
             return;
         }
 
+        $this->entityChangeSets = [];
         $this->entityInsertions = [];
         $this->entityUpdates = [];
     }
 
-    public function deferInsertions(array $entityInsertions, int $transactionNestingLevel): void
+    /**
+     * @inheritdoc
+     */
+    public function deferInsert(int $transactionNestingLevel, object $object, array $entityChangeSet): void
     {
         if ($this->enabled === false) {
             return;
         }
 
-        /** @var object[] $mergedEntityInsertions */
-        $mergedEntityInsertions = \array_merge(
-            $this->entityInsertions[$transactionNestingLevel] ?? [],
-            $entityInsertions
-        );
-        $this->entityInsertions[$transactionNestingLevel] = $mergedEntityInsertions;
+        $oid = \spl_object_hash($object);
+        $this->entityInsertions[$oid] = $object;
+        $this->entityChangeSets[$transactionNestingLevel][$oid] = $entityChangeSet;
     }
 
-    public function deferUpdates(array $entityUpdates, int $transactionNestingLevel): void
+    /**
+     * @inheritdoc
+     */
+    public function deferUpdate(int $transactionNestingLevel, object $object, array $entityChangeSet): void
     {
         if ($this->enabled === false) {
             return;
         }
 
-        /** @var object[] $mergedEntityUpdates */
-        $mergedEntityUpdates = \array_merge(
-            $this->entityUpdates[$transactionNestingLevel] ?? [],
-            $entityUpdates
-        );
-        $this->entityUpdates[$transactionNestingLevel] = $mergedEntityUpdates;
+        $oid = \spl_object_hash($object);
+        $this->entityUpdates[$oid] = $object;
+        $this->entityChangeSets[$transactionNestingLevel][$oid] = $entityChangeSet;
     }
 
     public function disable(): void
     {
         $this->enabled = false;
+        $this->clear();
```

Comment:
We don't have to clear it during disabling. Just imagine the following flow:

- Start a transaction.
- Create the entity A.
- Disable the dispatcher.
- Update the entity B.
- Enable the dispatcher.
- Commit the transaction.

This way we should dispatch an event about the entity A.

Reply:
Fixed.